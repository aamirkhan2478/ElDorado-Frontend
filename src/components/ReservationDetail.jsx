import React, { useState, Fragment } from 'react';
import PropType from 'prop-types';
import {
  Accordion,
  AccordionHeader,
  AccordionBody,
} from '@material-tailwind/react';
import { BookmarkIcon, BookmarkSlashIcon } from '@heroicons/react/24/solid';
import { TagIcon } from '@heroicons/react/24/outline';
import moment from 'moment';

const ReservationDetail = ({
  title, pickupDate, returnDate, model,
}) => {
  const [open, setOpen] = useState(0);

  const handleOpen = (value) => {
    setOpen(open === value ? 0 : value);
  };

  const customAnimation = {
    mount: { scale: 1 },
    unmount: { scale: 0.9 },
  };

  const formatDate = (date) => moment(date).format('ddd, MMM Do, YYYY');
  return (
    <>
      <Accordion open={open === 1} animate={customAnimation}>
        <AccordionHeader onClick={() => handleOpen(1)}>{title}</AccordionHeader>
        <AccordionBody>
          <ul className="list-outside list-disc">
            <li className="flex gap-1 items-center">
              <TagIcon className="w-7 stroke-amber-600" />
              <span>{model}</span>
            </li>
            <li className="flex gap-1 items-center">
              <BookmarkIcon className="w-7 fill-blue-gray-600" />
              <span>{formatDate(pickupDate)}</span>
            </li>
            <li className="flex gap-1 items-center">
              <BookmarkSlashIcon className="w-7 fill-red-500" />
              <span>{formatDate(returnDate)}</span>
            </li>
          </ul>
        </AccordionBody>
      </Accordion>
    </>
  );
};

ReservationDetail.propTypes = {
  title: PropType.string.isRequired,
  model: PropType.string.isRequired,
  pickupDate: PropType.string.isRequired,
  returnDate: PropType.string.isRequired,
};
export default ReservationDetail;
