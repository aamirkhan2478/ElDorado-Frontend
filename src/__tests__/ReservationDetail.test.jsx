import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/';
import ReservationDetail from '../components/ReservationDetail';

describe('Reservation Detail component', () => {
  test('Should render the proper Reservation Detail component', () => {
    const reservationDetail = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <ReservationDetail title pickupDate returnDate model />
          </Router>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(reservationDetail).toMatchSnapshot();
  });
});
