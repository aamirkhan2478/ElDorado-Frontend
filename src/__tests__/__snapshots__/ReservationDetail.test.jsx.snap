// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation Detail component Should render the proper Reservation Detail component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="block relative w-full"
      >
        <button
          class="flex justify-between items-center w-full py-4 border-b border-b-blue-gray-100 text-blue-gray-700 antialiased font-sans text-xl font-semibold leading-snug select-none hover:text-blue-gray-900 transition-colors"
          type="button"
        >
          <span
            class="w-5 h-5 ml-4"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 5a1 1 0 011 1v3h3a1 1 0 110 2h-3v3a1 1 0 11-2 0v-3H6a1 1 0 110-2h3V6a1 1 0 011-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
        <div
          class="overflow-hidden"
          style="height: 0px;"
        >
          <div
            class="block w-full py-4 text-gray-700 antialiased font-sans text-sm font-light leading-normal"
            style="opacity: 0; transform: scale(0.9) translateZ(0);"
          >
            <ul
              class="list-outside list-disc"
            >
              <li
                class="flex gap-1 items-center"
              >
                <svg
                  aria-hidden="true"
                  class="w-7 stroke-amber-600"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.568 3H5.25A2.25 2.25 0 003 5.25v4.318c0 .597.237 1.17.659 1.591l9.581 9.581c.699.699 1.78.872 2.607.33a18.095 18.095 0 005.223-5.223c.542-.827.369-1.908-.33-2.607L11.16 3.66A2.25 2.25 0 009.568 3z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M6 6h.008v.008H6V6z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span />
              </li>
              <li
                class="flex gap-1 items-center"
              >
                <svg
                  aria-hidden="true"
                  class="w-7 fill-blue-gray-600"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M6.32 2.577a49.255 49.255 0 0111.36 0c1.497.174 2.57 1.46 2.57 2.93V21a.75.75 0 01-1.085.67L12 18.089l-7.165 3.583A.75.75 0 013.75 21V5.507c0-1.47 1.073-2.756 2.57-2.93z"
                    fill-rule="evenodd"
                  />
                </svg>
                <span>
                  Invalid date
                </span>
              </li>
              <li
                class="flex gap-1 items-center"
              >
                <svg
                  aria-hidden="true"
                  class="w-7 fill-red-500"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.53 2.47a.75.75 0 00-1.06 1.06l18 18a.75.75 0 101.06-1.06l-18-18zM20.25 5.507v11.561L5.853 2.671c.15-.043.306-.075.467-.094a49.255 49.255 0 0111.36 0c1.497.174 2.57 1.46 2.57 2.93zM3.75 21V6.932l14.063 14.063L12 18.088l-7.165 3.583A.75.75 0 013.75 21z"
                  />
                </svg>
                <span>
                  Invalid date
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="block relative w-full"
    >
      <button
        class="flex justify-between items-center w-full py-4 border-b border-b-blue-gray-100 text-blue-gray-700 antialiased font-sans text-xl font-semibold leading-snug select-none hover:text-blue-gray-900 transition-colors"
        type="button"
      >
        <span
          class="w-5 h-5 ml-4"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 5a1 1 0 011 1v3h3a1 1 0 110 2h-3v3a1 1 0 11-2 0v-3H6a1 1 0 110-2h3V6a1 1 0 011-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <div
        class="overflow-hidden"
        style="height: 0px;"
      >
        <div
          class="block w-full py-4 text-gray-700 antialiased font-sans text-sm font-light leading-normal"
          style="opacity: 0; transform: scale(0.9) translateZ(0);"
        >
          <ul
            class="list-outside list-disc"
          >
            <li
              class="flex gap-1 items-center"
            >
              <svg
                aria-hidden="true"
                class="w-7 stroke-amber-600"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.568 3H5.25A2.25 2.25 0 003 5.25v4.318c0 .597.237 1.17.659 1.591l9.581 9.581c.699.699 1.78.872 2.607.33a18.095 18.095 0 005.223-5.223c.542-.827.369-1.908-.33-2.607L11.16 3.66A2.25 2.25 0 009.568 3z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M6 6h.008v.008H6V6z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span />
            </li>
            <li
              class="flex gap-1 items-center"
            >
              <svg
                aria-hidden="true"
                class="w-7 fill-blue-gray-600"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.32 2.577a49.255 49.255 0 0111.36 0c1.497.174 2.57 1.46 2.57 2.93V21a.75.75 0 01-1.085.67L12 18.089l-7.165 3.583A.75.75 0 013.75 21V5.507c0-1.47 1.073-2.756 2.57-2.93z"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Invalid date
              </span>
            </li>
            <li
              class="flex gap-1 items-center"
            >
              <svg
                aria-hidden="true"
                class="w-7 fill-red-500"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.53 2.47a.75.75 0 00-1.06 1.06l18 18a.75.75 0 101.06-1.06l-18-18zM20.25 5.507v11.561L5.853 2.671c.15-.043.306-.075.467-.094a49.255 49.255 0 0111.36 0c1.497.174 2.57 1.46 2.57 2.93zM3.75 21V6.932l14.063 14.063L12 18.088l-7.165 3.583A.75.75 0 013.75 21z"
                />
              </svg>
              <span>
                Invalid date
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
